pkg/sdk/external_tables.go

```diff
@@ -0,0 +1,472 @@
+	SetTag      []TagAssociation      `ddl:"keyword" sql:"SET TAG"`
+	UnsetTag    []ObjectIdentifier    `ddl:"keyword" sql:"UNSET TAG"`
+}
+
+type RefreshExternalTable struct {
+	Path string `ddl:"parameter,no_equals,single_quotes"`
+}
+
+type ExternalTableFile struct {
+	Name string `ddl:"keyword,single_quotes"`
+}
+
+// AlterExternalTablePartitionOptions based on https://docs.snowflake.com/en/sql-reference/sql/alter-external-table
+type AlterExternalTablePartitionOptions struct {
+	alterExternalTable bool                    `ddl:"static" sql:"ALTER EXTERNAL TABLE"`
+	IfExists           *bool                   `ddl:"keyword" sql:"IF EXISTS"`
+	name               AccountObjectIdentifier `ddl:"identifier"`
+	AddPartitions      []Partition             `ddl:"keyword,parentheses" sql:"ADD PARTITION"`
+	DropPartition      *bool                   `ddl:"keyword" sql:"DROP PARTITION"`
+	Location           string                  `ddl:"parameter,no_equals,single_quotes" sql:"LOCATION"`
+}
+
+type Partition struct {
+	ColumnName string `ddl:"keyword"`
+	Value      string `ddl:"parameter,single_quotes"`
+}
+
+// DropExternalTableOptions based on https://docs.snowflake.com/en/sql-reference/sql/drop-external-table
+type DropExternalTableOptions struct {
+	dropExternalTable bool                    `ddl:"static" sql:"DROP EXTERNAL TABLE"`
+	IfExists          *bool                   `ddl:"keyword" sql:"IF EXISTS"`
+	name              AccountObjectIdentifier `ddl:"identifier"`
+	DropOption        *ExternalTableDropOption
+}
+
+type ExternalTableDropOption struct {
+	Restrict *bool `ddl:"keyword" sql:"RESTRICT"`
+	Cascade  *bool `ddl:"keyword" sql:"CASCADE"`
+}
+
+// ShowExternalTableOptions based on https://docs.snowflake.com/en/sql-reference/sql/show-external-tables
+type ShowExternalTableOptions struct {
+	show           bool       `ddl:"static" sql:"SHOW"`
+	Terse          *bool      `ddl:"keyword" sql:"TERSE"`
+	externalTables bool       `ddl:"static" sql:"EXTERNAL TABLES"`
+	Like           *Like      `ddl:"keyword" sql:"LIKE"`
+	In             *In        `ddl:"keyword" sql:"IN"`
+	StartsWith     *string    `ddl:"parameter,single_quotes,no_equals" sql:"STARTS WITH"`
+	LimitFrom      *LimitFrom `ddl:"keyword" sql:"LIMIT"`
+}
+
+type externalTableRow struct {
+	CreatedOn           time.Time
+	Name                string
+	DatabaseName        string
+	SchemaName          string
+	Invalid             bool
+	InvalidReason       sql.NullString
+	Owner               string
+	Comment             string
+	Stage               string
+	Location            string
+	FileFormatName      string
+	FileFormatType      string
+	Cloud               string
+	Region              string
+	NotificationChannel sql.NullString
+	LastRefreshedOn     sql.NullTime
+	TableFormat         string
+	LastRefreshDetails  sql.NullString
+	OwnerRoleType       string
+}
+
+func (e externalTableRow) ToExternalTable() ExternalTable {
```

Comment:
can we do var _ con... = (*externalTableRow)(nil) to avoid allocating memory ? I'm not sure if it will, because maybe it should be optimized out of final binary, but still, wdyt ?

Reply:
I like the `new` syntax much more TBH, but either will work, so I don't care that much which

Reply:
It looks like `new` doesn't allocate in global interface assertions, so I can change it for simplicity and consistency with other files

Reply:
Well, we are using both :) For validatable we have sh*tload of `new()` :)

Reply:
(you can close after reading)